state/execution.go

```diff
@@ -425,7 +426,11 @@ func updateState(
 // Fire TxEvent for every tx.
 // NOTE: if Tendermint crashes before commit, some or all of these events may be published again.
 func fireEvents(logger log.Logger, eventBus types.BlockEventPublisher, block *types.Block, abciResponses *ABCIResponses) {
-	eventBus.PublishEventNewBlock(types.EventDataNewBlock{block})
+	eventBus.PublishEventNewBlock(types.EventDataNewBlock{
+		Block:            block,
+		ResultBeginBlock: *abciResponses.BeginBlock,
```

Comment:
Not sure we want to bundle begin/end block results together with the block. The original issue was more about notifying clients about tags in begin block / commit. Blocks can be quite large.

@t-bast @ebuchman thoughts?

Reply:
Yeah, that is why I mentioned we need to discuss the design :-)

So the other solution would be a new event type for these.

Reply:
I agree with @melekes. In my scenario I really don't need the whole result, including tags (that I can somehow control from my ABCI app) are enough.

Reply:
It looks like this is going in the right direction, it's just a matter of finding a lightweight way of passing the tags without the block right?

Reply:
Yes, if clients don't want to subscribe to full blocks we could just emit an additional (new event type or we could use `EventNewBlockHeader`?) event that doesn't contain the block, but still contains the tags and/or other metadata.

Reply:
I think the most correct way would be to emit tags in both `EventNewBlock` and `EventNewBlockHeader` and then clients can decide which to listen on.

Reply:
I agree with your latest comment. I'll let the Tendermint team comment on the design choices.